calculator_test.go

```diff
@@ -2,23 +2,202 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
+	"time"
 )
 
+// testCase represents a case for testing non-error producing Calculator functions.
+type testCase struct {
```

Comment:
This is fine, but it's usual in tests to declare the test case struct type inside the test function, since it's unlikely to be useful anywhere else. (It so happens here that it might be, but let's stick to the standard pattern for now.) 

In fact, we don't even bother to give the struct type a name, because we don't need to refer to it; instead, we define a slice of test cases, and then just loop over them. Something like this:

```go
tcs := []struct{
	a, b...
}{
	{ a: 1, b: 2... },
}
```

Reply:
I submitted this code before I read Dave Cheney's _[Prefer table driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests)_, so your feedback makes even more sense now!